### PR TITLE
added Homework 1

### PR DESCRIPTION
**Pull request by nevin75**

### CODE REVIEW

File: Src/BootCamp.Chapter/Program.cs [hunk 1]
```diff
@@ -6,7 +6,63 @@ class Program
...
+                "their weight is " + weight + " kg " +
+                "and their height is " + height + " cm.");
+        }
+
+        private static void ReportBMI(string firstName, double weight, double height)
+        {
+            Console.WriteLine("\n" + firstName + "'s BMI is " + (CalculateBMI(weight, height)) + ".");
+        }
+
+        private static double CalculateBMI(double weight, double height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Program.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880&bid=16045032)

File: Src/BootCamp.Chapter/Program.cs [hunk 2]
```diff
@@ -6,7 +6,63 @@ class Program
...
+
+        private static void ReportBasicInfo(string firstName, string surname, int age, double weight, double height)
+        {
+            Console.WriteLine("\n" + firstName + " " + surname +
+                " is " + age + " years old, " +
+                "their weight is " + weight + " kg " +
+                "and their height is " + height + " cm.");
+        }
+
+        private static void ReportBMI(string firstName, double weight, double height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Rename method 'ReportBMI' to match pascal case naming rules, consider using 'ReportBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Program.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880&bid=16045032)

File: Src/BootCamp.Chapter/Program.cs [hunk 3]
```diff
@@ -6,7 +6,63 @@ class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            char again = 'y';
```
Comment (diverges) [inline Src/BootCamp.Chapter/Program.cs]: This is what a sample comment would look like in a PR.

File: Src/BootCamp.Chapter/Program.cs [hunk 4]
```diff
@@ -6,7 +6,63 @@ class Program
...
+
+        private static double CalculateBMI(double weight, double height)
+        {
+            double heightInMeters = height * .01;
+            return weight / (heightInMeters * heightInMeters);
+        }
+
+        private static char RequestUserChoiceToRepeatOrNot()
+        {
+            Console.Write("\nWould you like to repeat this program? (y or n)");
```
[the remark this answers is not in the document]
Comment (nevin75) [reply Src/BootCamp.Chapter/Program.cs]: Thank you. I wanted a space between the earlier reporting and the question about repeating the program. Is there a standard way? A cleaner way?

From your comment, I think, within Main, after ReportBmi, perhaps I just put an additional Console.Writeline(""); in there to add in a newline.

**Comment by nevin75** [comment]
Did I "target the wrong branch"? It says "nevin 75 wants to merge 2 commits into `csinn:master` from `nevin75:master`

I think it should more specifically seeking to merge just the sub-set of the the Homework 1 and 2. Given that all I changed was a file in the target branch, is this just semantics? Am I actually changing/complicating more than I should?

Is it too late to repair this?

**Comment by nevin75** [comment]
Kai suggested Mentors give feedback on comments in the homework. Please consider if there's anything I should have done. My hope is that my naming makes comments redundant, but please let me know if I've missed where a comment should commonly be used.

**Comment by nevin75** [comment]
Naming: I'm unresolved with ReportBmi/ReportBMI. If I called it "ReportBodyMassIndex" is that better? Am I making it less efficient?

**Comment by nevin75** [comment]
If this wasn't mid-review, I'd also change "height" to "heightInCentimeters" and "weight" to "weightInKilograms." What is an appropriate protocol if I think of improvements after a pull request is partially reviewed?
[Sorry if this is spamming. I chose to separate each question for easier response.]

**Comment by simonasB** [comment]
> Naming: I'm unresolved with ReportBmi/ReportBMI. If I called it "ReportBodyMassIndex" is that better? Am I making it less efficient?

I think the most expressive is ReportBodyMassIndex but it is fine to use other suggested names if you write a short comment what that BMI abbreviation means. But in general code should be self-descriptive as much as possible.

**Comment by simonasB** [comment]
> If this wasn't mid-review, I'd also change "height" to "heightInCentimeters" and "weight" to "weightInKilograms." What is an appropriate protocol if I think of improvements after a pull request is partially reviewed?
> [Sorry if this is spamming. I chose to separate each question for easier response.]

I would say go for the changes, improving the code is always a good idea (specially when learning). Just add a commit message describing why it has been done (e.g., improve readability).

**Comment by simonasB** [comment]
> Did I "target the wrong branch"? It says "nevin 75 wants to merge 2 commits into `csinn:master` from `nevin75:master`
> 
> I think it should more specifically seeking to merge just the sub-set of the the Homework 1 and 2. Given that all I changed was a file in the target branch, is this just semantics? Am I actually changing/complicating more than I should?
> 
> Is it too late to repair this?

You should target csinn:Chapter1/Homework/1And2 branch. It is not too late, just try to change a base branch. It is possible using Github when you click Edit.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Program.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880&bid=16045032)

File: README.md [hunk 5]
```diff
@@ -1,4 +1,77 @@
...
+  Lesson/LessonName  
+  Homework/LessonName  
+  ```
+  ## Git
+  Fork this repository (Homework/LessonName). 
+  The fork target is called upstream branch. For every chapter, you will need to keep updating upstream branch. 
+  How to work with forked repository in Visual Studio Teams:  
+  http://doc.fireflymigration.com/working-with-github-fork-in-visual-studio.html
+  
+  ## How to complete homework?
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-heading-punctuation] Don’t add a trailing `?` to headings](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 6]
```diff
@@ -1,4 +1,77 @@
...
+building a house if you never built one. Programming is hard only until you practice it (like any other skill). I would like to invite you to learn programming and C# following this course to practice programming and ignite passion for finding little miracles in code every day 🙂
+  
+  # Requirements
+  Visual Studio Community 2019 or similar with Web Development and desktop development workloads.  
+
+  Patience and persistence  
+  Git installed  
+  Github account  
+  
+  # Homework
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-multiple-toplevel-headings] Don’t use multiple top level headings (1:1)](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 7]
```diff
@@ -1,4 +1,77 @@
...
+  ### Finishing homework
+  You can consider your homework complete if all of the above steps pass. 
+  **Don't merge the PR to master branch**
+  A single best homework implementation will be merged to master. 
+
+  
+
+  # Community
+  If you want to join as either mentor or pupil or have any questions, please refer to our discord Community.  
+  https://discord.gg/rCMKcUU
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-consecutive-blank-lines] Remove 9 lines after node](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 8]
```diff
@@ -1,4 +1,77 @@
...
+  Code reviews on GitHub- Todo: link
+  
+  ### Finishing homework
+  You can consider your homework complete if all of the above steps pass. 
+  **Don't merge the PR to master branch**
+  A single best homework implementation will be merged to master. 
+
+  
+
+  # Community
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-consecutive-blank-lines] Remove 10 lines after node](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-multiple-toplevel-headings] Don’t use multiple top level headings (1:1)](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 9]
```diff
@@ -1,4 +1,77 @@
...
-C# boot camp
+[![Build Status](https://dev.azure.com/almantusk/From-Zero-To-Hero/_apis/build/status/csinn.CSharp-From-Zero-To-Hero?branchName=master)](https://dev.azure.com/almantusk/From-Zero-To-Hero/_build/latest?definitionId=6&branchName=master)
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/a358adf21c2442cd91a4827a50213ff1)](https://www.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=csinn/CSharp-From-Zero-To-Hero&amp;utm_campaign=Badge_Grade)
+  # The vision
+"Programming is hard". Yes.But not harder than running a marathon for a person has never run. It's not harder than 
+building a house if you never built one. Programming is hard only until you practice it (like any other skill). I would like to invite you to learn programming and C# following this course to practice programming and ignite passion for finding little miracles in code every day 🙂
+  
+  # Requirements
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-multiple-toplevel-headings] Don’t use multiple top level headings (1:1)](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 10]
```diff
@@ -1,4 +1,77 @@
 # CSharp-From-Zero-To-Hero
-[![Build Status](https://dev.azure.com/almantusk/From-Zero-To-Hero/_apis/build/status/Almantask.CSharp-From-Zero-To-Hero?branchName=master)](https://dev.azure.com/almantusk/From-Zero-To-Hero/_build/latest?definitionId=5&branchName=master)
-[![Codacy Badge](https://api.codacy.com/project/badge/Grade/a358adf21c2442cd91a4827a50213ff1)](https://www.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=Almantask/CSharp-From-Zero-To-Hero&amp;utm_campaign=Badge_Grade)  
-C# boot camp
+[![Build Status](https://dev.azure.com/almantusk/From-Zero-To-Hero/_apis/build/status/csinn.CSharp-From-Zero-To-Hero?branchName=master)](https://dev.azure.com/almantusk/From-Zero-To-Hero/_build/latest?definitionId=6&branchName=master)
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/a358adf21c2442cd91a4827a50213ff1)](https://www.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=csinn/CSharp-From-Zero-To-Hero&amp;utm_campaign=Badge_Grade)
+  # The vision
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[no-multiple-toplevel-headings] Don’t use multiple top level headings (1:1)](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

File: README.md [hunk 11]
```diff
@@ -1,4 +1,77 @@
...
+  ## Git
+  Fork this repository (Homework/LessonName). 
+  The fork target is called upstream branch. For every chapter, you will need to keep updating upstream branch. 
+  How to work with forked repository in Visual Studio Teams:  
+  http://doc.fireflymigration.com/working-with-github-fork-in-visual-studio.html
+  
+  ## How to complete homework? 
+  ### Implementation of homework
+  For the first chapters, homework branch will come with 2 projects:
+  ```
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[fenced-code-flag] Missing code-language flag](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4828880)

**Comment by nevin75** [comment]
I think I'm stuck. In my last commit & push, I modified the pull request to improve naming of some functions and variables. I don't think that created any issues.

However, I also tried to target the right branch for the pull request. Instead of targeting csinn:Master, I targeted the Homework branch.
image

Perhaps that changed something? Codacy is now rejecting a bunch of stuff related to a Readme file. I don't know what happened and I'm not sure what to do.

**Comment by Almantask** [comment]
> I think I'm stuck. In my last commit & push, I modified the pull request to improve naming of some functions and variables. I don't think that created any issues.
> 
> However, I also tried to target the right branch for the pull request. Instead of targeting csinn:Master, I targeted the Homework branch.
> image
> 
> Perhaps that changed something? Codacy is now rejecting a bunch of stuff related to a Readme file. I don't know what happened and I'm not sure what to do.

You should target the homework branch. Ignore codacy for now :)

File: Src/BootCamp.Chapter/Program.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Almantask) [reply Src/BootCamp.Chapter/Program.cs]: If you want to put an extra line, you can do what you do, but just use WriteLine and not \n before every statement. So for first WriteLine"\n" is okay- for others- you won't have to use it anymore.

**Pull request closed without merging**